sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpError.cs

```diff
@@ -40,19 +30,194 @@ internal static class AmqpError
+            {
+                return new QuotaExceededException(eventHubsResource, description);
+            }
+
+            // The service does not understand how to process the request.
+
+            if (String.Equals(condition, AmqpErrorCode.NotAllowed.Value, StringComparison.InvariantCultureIgnoreCase))
+            {
+                return new InvalidOperationException(description);
+            }
+
+            if (String.Equals(condition, AmqpErrorCode.NotImplemented.Value, StringComparison.InvariantCultureIgnoreCase))
+            {
+                return new NotSupportedException(description);
+            }
+
+            // The Event Hubs resource was not valid or communication with the service was interrupted.
+
+            if (String.Equals(condition, AmqpErrorCode.NotFound.Value, StringComparison.InvariantCultureIgnoreCase))
+            {
+                if (NotFoundExpression.IsMatch(description)
+                    || (description.IndexOf(NotFoundStatusText, StringComparison.InvariantCultureIgnoreCase) >= 0))
+                {
+                    return new EventHubsResourceNotFoundException(eventHubsResource, description);
+                }
+
+
+                return new EventHubsCommunicationException(eventHubsResource, description);
+            }
+
+            // There was no specific exception that could be determined; fall back to a generic one.
+
+            return new EventHubsException(true, eventHubsResource, description);
+        }
+
+        /// <summary>
+        ///   Determines the applicable error condition for a give response message.
+        /// </summary>
+        ///
+        /// <param name="response">The AMQP response message to consider.</param>
+        ///
+        /// <returns>The AMQP error condition that best represents the response.</returns>
+        ///
+        private static AmqpSymbol DetermineErrorCondition(AmqpMessage response)
+        {
+            AmqpSymbol condition;
+
+            // If there was an error condition present, use that.
+
+            if (response.ApplicationProperties.Map.TryGetValue<AmqpSymbol>(AmqpResponse.ErrorCondition, out condition))
+            {
+                return condition;
+            }
 
-        /// <summary>Indicates that the requested Event Hubs resource is disabled.</summary>
-        public static readonly AmqpSymbol ResourceDisabledError = AmqpConstants.Vendor + ":entity-disabled";
+            // If no error condition was present, perform a reverse lookup in the mappings to determine the
+            // condition from the response status code.
 
-        /// <summary>Indicates that the producer requesting an operation is not allowed to publish events to the requested resource.</summary>
-        public static readonly AmqpSymbol PublisherRevokedError = AmqpConstants.Vendor + ":publisher-revoked";
+            if ((response.ApplicationProperties.Map.TryGetValue<AmqpResponseStatusCode>(AmqpResponse.StatusCode, out var statusCode))
```

Comment:
This has to change to pulling out an `int` code and then casting it.   Testing for conversion showed that the direct attempt for the enum was never successful.